### PR TITLE
[pytorch] | [benchmark] PyTorch p3dn ImageNet Benchmark

### PR DESCRIPTION
**Pull request by YYStreet**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [pytorch] | [benchmark]

*Description:*
Add PyTorch p3dn ImageNet Benchmark 

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py [hunk 1]
```diff
@@ -2,15 +2,40 @@
...
 
 
 PT_PERFORMANCE_TRAINING_GPU_SYNTHETIC_CMD = os.path.join(CONTAINER_TESTS_PREFIX, "benchmark", "run_pytorch_training_performance_gpu_synthetic")
+PT_PERFORMANCE_TRAINING_GPU_IMAGENET_CMD = os.path.join(CONTAINER_TESTS_PREFIX, "benchmark", "run_pytorch_training_performance_gpu_imagenet")
 
 PT_EC2_GPU_SYNTHETIC_INSTANCE_TYPE = "p3.16xlarge"
+PT_EC2_GPU_IMAGENET_INSTANCE_TYPE = "p3dn.24xlarge"
 
 
 @pytest.mark.parametrize("ec2_instance_type", [PT_EC2_GPU_SYNTHETIC_INSTANCE_TYPE], indirect=True)
```
Comment (YYStreet) [inline test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:16]: This test is to be as a comparison to verify this env works fine with the test using common AMI

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+       --workers 30 \
+       --epochs 0 \
+       ~/imagenet
+END=$(date +%s)
+DIFF=$(( END - START ))
+echo Preparation Stage took $(( DIFF/60 )) mins to finish>&2
+
+echo Benchmark Started: >&2
+START=$(date +%s)
+timeout 60m python examples/imagenet/main.py \
```
Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:37]: Will adjust to 100m for mainline pipeline

File: test/testrunner.py [hunk 3]
```diff
@@ -97,8 +97,10 @@ def main():
     all_image_list = dlc_images.split(" ")
     standard_images_list = [image_uri for image_uri in all_image_list if "example" not in image_uri]
     eks_cluster_name = None
-    benchmark_mode = "benchmark" in test_type
-    specific_test_type = re.sub("benchmark-", "", test_type) if benchmark_mode else test_type
+    # benchmark_mode = "benchmark" in test_type
```
Comment (YYStreet) [inline test/testrunner.py:100]: Config test to have benchmark script tested under PR context. Will be reverted before merging in

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+BIN_DIR=${HOME_DIR}/bin
+LOG_DIR=${HOME_DIR}/logs/gpu
+LOG_FILE=image_results_${COMMIT_INFO}_${TIMESTAMP}.txt
+
+mkdir -p ${HOME_DIR}
+mkdir -p ${BIN_DIR}
+mkdir -p ${LOG_DIR}
+
+cd ~
+echo Benchmark Preparation started: >&2
```
Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:19]: To run the script twice and exit forcely for the 1st round is to avoid the stuck of the testing script, which is likely to happen on a new launched AMI. After that, 2nd 2round would run smoothly

File: test/dlc_tests/conftest.py [hunk 5]
```diff
@@ -114,7 +114,8 @@ def ec2_instance(
         region = P3DN_REGION
         ec2_client = boto3.client("ec2", region_name=region, config=Config(retries={"max_attempts": 10}))
         ec2_resource = boto3.resource("ec2", region_name=region, config=Config(retries={"max_attempts": 10}))
-        ec2_instance_ami = UBUNTU_16_BASE_DLAMI_US_EAST_1
+        if ec2_instance_ami == UBUNTU_16_BASE_DLAMI_US_WEST_2:
```
Comment (arjkesh) [inline test/dlc_tests/conftest.py:117]: why is this line needed?

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+       -a resnet50 \
+       --dist-url 'tcp://127.0.0.1:8080' \
+       --dist-backend 'nccl' \
+       --multiprocessing-distributed \
+       --world-size 1 \
+       --rank 0 \
+       --workers 30 \
+       --epochs 16 \
+       ~/imagenet 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
+python /test/bin/benchmark/get_pytorch_performance_num.py ${LOG_DIR}/"${LOG_FILE}" >> ${LOG_DIR}/"${LOG_FILE}"
```
Comment (arjkesh) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:47]: Is there anywhere where we define a failure condition or threshold for this test?

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py [hunk 7]
```diff
@@ -2,16 +2,50 @@
...
 
 
 @pytest.mark.model("resnet50")
 @pytest.mark.parametrize("ec2_instance_type", [PT_EC2_GPU_SYNTHETIC_INSTANCE_TYPE], indirect=True)
 def test_performance_pytorch_gpu_synthetic(pytorch_training, ec2_connection, gpu_only, py3_only):
     execute_ec2_training_performance_test(ec2_connection, pytorch_training, PT_PERFORMANCE_TRAINING_GPU_SYNTHETIC_CMD)
+
+
+@pytest.mark.model("resnet50")
+@pytest.mark.parametrize("ec2_instance_ami", [PT_GPU_PY3_BENCHMARK_IMAGENET_AMI], indirect=True)
```
Comment (arjkesh) [inline test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:24]: why is a different AMI needed from regular us-east-1 AMI?

File: test/dlc_tests/conftest.py (hunk 5 above)
Comment (YYStreet) [reply test/dlc_tests/conftest.py:117]: This test is using a specific AMI https://github.com/YYStreet/deep-learning-containers/blob/pt/test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py#L24

The origin logic is to forcefully apply `UBUNTU_16_BASE_DLAMI_US_EAST_1` to any tests running in us-east-1 and ignore the arg of AMI, which is incorrect

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py (hunk 7 above)
Comment (YYStreet) [reply test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:24]: The benchmark test toolkit and data set is pre-configured in this AMI. I am following the practice of manual steps for running this benchmark.

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet (hunk 6 above)
Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:47]: Will address it separately

File: test/test_utils/__init__.py [hunk 8]
```diff
@@ -25,7 +25,8 @@
 # Deep Learning Base AMI (Ubuntu 16.04) Version 25.0 used for EC2 tests
 UBUNTU_16_BASE_DLAMI_US_WEST_2 = "ami-0e5a388144f62e4f5"
 UBUNTU_16_BASE_DLAMI_US_EAST_1 = "ami-0da7f2daf5e92c6f2"
-UL_AMI_LIST = [UBUNTU_16_BASE_DLAMI_US_WEST_2, UBUNTU_16_BASE_DLAMI_US_EAST_1]
+PT_GPU_PY3_BENCHMARK_IMAGENET_AMI = "ami-0673bb31cc62485dd"
+UL_AMI_LIST = [UBUNTU_16_BASE_DLAMI_US_WEST_2, UBUNTU_16_BASE_DLAMI_US_EAST_1, PT_GPU_PY3_BENCHMARK_IMAGENET_AMI]
```
Comment (nskool) [inline test/test_utils/__init__.py:29]: PT_GPU_PY3_BENCHMARK_IMAGENET_AMI: Is this AMI in us-east-1 or us-east-2?

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet (hunk 4 above)
Comment (nskool) [reply test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:19]: Can we avoid duplication here by parameterizing the timeout?

File: test/dlc_tests/conftest.py (hunk 5 above)
Comment (nskool) [reply test/dlc_tests/conftest.py:117]: This condition is confusing. It checks if the AMI is us-west-2, and then assigns a us-east-1 AMI instead. Since we expect default to be us-east-1  - it's better to assign the default unless the AMI is the same as PT AMI.

File: test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import argparse
```
Comment (nskool) [inline test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py:1]: Not sure if this should be a sepate file. If we're creating a separate file - we should create a generic 'value checker file' that could be used across different benchmarking test. But then again - such a function is better located in conftest or utils, depending on relevance to pytest.

File: test/test_utils/__init__.py (hunk 8 above)
Comment (YYStreet) [reply test/test_utils/__init__.py:29]: It is in us-east-1. I will add the region info in its name

File: test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py (hunk 9 above)
Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py:1]: For the 1st question, I think your suggestion works. However if we need to use other AMIs located in us-east-1 for other tests, we need to change this line another time.

Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py:1]: 1st parameterizing issue will be resolved in next commit

Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/benchmark/get_pytorch_performance_num.py:1]: For issue 3, there is another python helper function script located in this folder for post-processing. I might work in a separate PR for them at the same time

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+LOG_FILE=image_results_${COMMIT_INFO}_${TIMESTAMP}.txt
+
+mkdir -p ${HOME_DIR}
+mkdir -p ${BIN_DIR}
+mkdir -p ${LOG_DIR}
+
+function trigger_test() {
+  time_out=$1
+  epochs=0
+  if [ "$2" == "run" ]
```
Comment (nskool) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:21]: Please make the condition for the other type: "prep" more explicit by checking if [ "$2" == "prep" ]
